Dedicated SKU column (for issue #495) (#516)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/columns/skumeter.md

# SKU Meter

The SKU Meter describes the capability being metered or measured by a particular SKU in a charge.

Providers often have billing models in which multiple SKUs exist for a given service to describe and bill for different capabilities for that service. For example, an object storage service may have separate SKUs for capabilities such as object storage, API requests, data transfer, encryption, object management. This field helps practitioners understand which capabilities are being metered by SKUs that appear in a FOCUS dataset.

> The SkuMeter column MUST be present in the billing data when a provider supports describing the capabilities contained within a SKU. The column MUST be of type String and MAY be null.

Reviewer: I suggest we tighten up the requirements and align the nullability with SkuId (I believe all Providers can describe the capability of the component they are measuring and charging). 

Should we opt for that, even though for some other columns we avoided referencing the SkuId and/or SkuPriceId columns, I believe that, just like we referenced SkuPriceId in the case of SkuPriceDetails, it is okay to reference SkuId in the case of SkuMeter.

Adding quote from SkuPriceDetails for reference:

> The SkuPriceDetails column MUST be present in the billing data when the provider includes a SkuPriceId.

```suggestion
The SkuMeter column adheres to the following requirements:

 * SkuMeter MUST be present in the billing data when when the provider includes a [SkuId](#skuid).
 * SkuMeter MUST be of type String.
 * SkuMeter MUST be null when SkuId is null and MUST NOT be null when SkuId is not null.
```

Author: For AWS, not every SKU will have a MeterName (i.e. usage type). I'd be happy to say this should be null when SkuId is null, but I wouldn't want to force a value when SkuId isn't null.

## Column ID

SkuMeter

## Display Name

> SkuMeter

Reviewer: ```suggestion
SKU Meter
```

Author: fixed and updated to SKU Meter Name, thanks!

# SKU Meter

The SKU Meter describes the functionality being metered or measured by a particular SKU in a charge.

> Providers often have billing models in which multiple SKUs exist for a given service to describe and bill for different functionalities for that service. For example, an object storage service may have separate SKUs for functionalities such as object storage, API requests, data transfer, encryption, object management. This field helps practitioners understand which functionalities are being metered by the different SKUs that appear in a FOCUS dataset.

Reviewer 2: ```suggestion
Providers often have billing models in which multiple SKUs exist for a given service to describe and bill for different functionalities for that service. For example, an object storage service may have separate SKUs for functionalities such as object storage, API requests, data transfer, encryption, and object management. This field helps practitioners understand which functionalities are being metered by the different SKUs that appear in a FOCUS dataset.
```

Adding "and" to the list.

The SkuMeter column adheres to the following requirements:

* SkuMeter MUST be present in the billing data when when the provider includes a [SkuId](#skuid).
* SkuMeter MUST be of type String.
* SkuMeter MUST be null when SkuId is null.
* SkuMeter SHOULD NOT be null when SkuId is not null.

> * SkuMeterName SHOULD remain consistent over time for a given SkuId.

Reviewer 2: References "SkuMeterName" vs "SkuMeter" though I know that the naming is up for debate currently

Author: thanks - fixed
